src/Microsoft.Health.Fhir.Core/Features/Operations/BulkUpdate/BulkUpdateOrchestratorJob.cs

```diff
@@ -0,0 +1,223 @@
+                            definitions.Add(processingDefinition);
+                        }
+
+                        startId++; // make sure we do not intersect ranges
+
+                        rows = definitions.Count;
+                        if (rows > 0)
+                        {
+                            _logger.LogJobInformation(jobInfo, "Enqueuing bulk update job (1).");
+                            await _queueClient.EnqueueAsync(QueueType.BulkUpdate, cancel, groupId: jobInfo.GroupId, definitions: definitions.ToArray());
+                            atLeastOneWorkerJobRegistered = true;
+                        }
+                    }
+                });
+
+                if (!atLeastOneWorkerJobRegistered)
+                {
+                    _logger.LogJobInformation(jobInfo, "Creating bulk update definition (2).");
+                    var processingRecord = CreateProcessingDefinition(definition, searchService.Value, cancellationToken, definition.Type, null, true);
+
+                    _logger.LogJobInformation(jobInfo, "Enqueuing bulk update job (2).");
+
+                    await _queueClient.EnqueueAsync(QueueType.BulkUpdate, cancellationToken, groupId: jobInfo.GroupId, definitions: processingRecord);
+                }
+            }
+            else if (definition.IsParallel && !string.IsNullOrEmpty(definition.Type))
+            {
+                // read all the data from conditional search using Continuation token and create processing definitions and enqueue them by storing continuation token as well
+                IReadOnlyCollection<SearchResultEntry> searchResults;
+                string ct;
+                string ict;
+                string prevCT = null;
+                var searchParams = definition.SearchParameters.ToList();
+                (searchResults, ct, ict) = await searchService.Value.ConditionalSearchAsync(
+                    definition.Type,
+                    searchParams,
+                    cancellationToken,
+                    (int?)definition.MaximumNumberOfResourcesPerQuery,
+                    versionType: ResourceVersionType.Latest,
+                    onlyIds: true,
+                    logger: _logger);
+
+                while (searchResults.Any() || !string.IsNullOrEmpty(prevCT))
+                {
+                    _logger.LogJobInformation(jobInfo, "Creating bulk update definition (3).");
+                    var processingRecord = CreateProcessingDefinition(definition, searchService.Value, cancellationToken, definition.Type, prevCT, false);
+
+                    _logger.LogJobInformation(jobInfo, "Enqueuing bulk update job (3).");
+                    await _queueClient.EnqueueAsync(QueueType.BulkUpdate, cancellationToken, groupId: jobInfo.GroupId, definitions: processingRecord);
+
+                    if (ct is null && ict is null)
```

Comment:
This is dangerous. If the last page of results (IE: ct is null) has an includes continuation token this will cause the search to run again, but with null as the continuation token. This will get the first page of results again and cause the job creation to enter an infinite loop.